arrows/serialize/protobuf/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+# Build / Install plugin for serialization
+
+find_package(Protobuf MODULE REQUIRED)
+
+if(NOT PROTOBUF_FOUND)
+  message(FATAL_ERROR "-- Protobuf is required!")
+endif()
+
+set( headers_public
+  bounding_box.h
+  detected_object_type.h
+  detected_object.h
+  detected_object_set.h
+  )
+
+set( sources
+  bounding_box.cxx
+  detected_object_type.cxx
+  detected_object.cxx
+  detected_object_set.cxx
+  )
+
+link_directories(${PROTOBUF_LIBRARY})
+include_directories(${PROTOBUF_INCLUDE_DIR})
+add_compile_options( -include "${CMAKE_BINARY_DIR}/vital/types/protobuf/vital_types_protobuf_export.h" )
```

Comment:
I am not 100% sure what this command is providing, but I see it didn't appear in the json CMakeLists.txt. Should it be there?

Reply:
# Need to add this include file to the compile because of a defficiency in the protobuf
# compiler. The compiler has the option to insert an export macro into class definitions
# but has no way of defining that macro. as of libprotoc 2.6.1 this is the accepted way
# of defining the export symbols.